src/components/Toggle/Toggle.tsx

```diff
@@ -0,0 +1,51 @@
+'use client';
+
+import clsx from 'clsx';
+import { FC, InputHTMLAttributes, forwardRef, useRef, useState } from 'react';
+import styles from './Toggle.module.css';
+
+type Props = {
+  /**
+   * 現在の状態が選択中かどうか
+   */
+  checked?: boolean;
+  /**
+   * 初期状態が選択状態かどうか
+   * @default false
+   */
+  defaultChecked?: boolean;
+  /**
+   * 値が変化した場合のコールバック
+   */
+  onChange?: InputHTMLAttributes<HTMLInputElement>['onChange'];
+  /**
+   * 無効状態かどうか
+   * @default false
+   */
+  disabled?: boolean;
+} & Omit<InputHTMLAttributes<HTMLInputElement>, 'children' | 'onChange'>;
+
+export const Toggle: FC<Props> = forwardRef<HTMLLabelElement, Props>(
+  ({ checked: checkedProps, defaultChecked, onChange, ...otherProps }, ref) => {
+    const { current: isUnControlled } = useRef(checkedProps === undefined);
+    const [isUnControlledChecked, setIsUnControlledChecked] = useState(defaultChecked);
+
+    const isChecked = isUnControlled ? isUnControlledChecked : checkedProps;
+
+    const handleChange: React.ChangeEventHandler<HTMLInputElement> = (event) => {
+      if (isUnControlled) {
+        setIsUnControlledChecked(event.target.checked);
+      }
+      if (onChange) onChange(event);
+    };
+
+    return (
+      <label ref={ref} className={clsx(styles.root, isChecked ? styles.on : styles.off)}>
+        <input type="checkbox" checked={isChecked} onChange={handleChange} className={styles.input} {...otherProps} />
```

Comment:
```suggestion
        <input type="checkbox"  role="switch" checked={isChecked} onChange={handleChange} className={styles.input} {...otherProps} />
```

I think we should use "switch" role.
fyi: https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch-checkbox/

Reply:
Thanks! 
Makes sense. but I don't know it is not used on major component libraries like [MUI](https://mui.com/material-ui/react-switch/) and [ChakuraUI](https://chakra-ui.com/docs/components/switch/usage) which I used as reference.

Reply:
with the `role='switch'`, I get the linter error anyway. 

without `aria-checked`, it throws `ロール「switch」では、ARIAステート「aria-checked」が必要です` eslint error
with `aria-checked`, it throws `要素「input」では、ARIAステート「aria-checked」は使用するべきではありません。使用する場合は、属性「checked」を追加してください`

can i ignore one of these ?

Reply:
hmmmm...
`ロール「switch」では、ARIAステート「aria-checked」が必要です ` does not seem to be in accordance with the specification. We should ignore this error.
https://github.com/w3c/html-aria/issues/227

Reply:
We should probably create an Issue, but I don't think it is likely to be fixed since it doesn't seem to be very actively maintained.
https://github.com/jsx-eslint/eslint-plugin-jsx-a11y

Reply:
Agreed 👍

The linter issue was from [markuplint](https://github.com/markuplint/markuplint/discussions/1079).

There is no inline ignore method like`eslint-disable` for `markuplint`  (ref: https://github.com/markuplint/markuplint/discussions/1079), so I think it is the only way to ignore by [css selectors](https://github.com/markuplint/markuplint/blob/7f949a0ab488c0045185fa14080c3ce671956afc/website/docs/guides/selectors.md). https://github.com/ubie-oss/ubie-ui/pull/18/commits/21e840dbc30f99805ecba37a35ce8c8320b235f6

As Downside, it will ignore all assertions related to `wai-aria` rule other than `Require the "aria-checked" ARIA state on the "switch" role` but I don't think there is better way without improvement of `markuplint` itself.😭

Reply:
As a side note, markuplint is currently `v4.3.0`, while we use `v3.12.0`.
Maybe we should upgrade. 👀

Reply:
and I created the [issue](https://github.com/markuplint/markuplint/issues/1517).

Reply:
Oh, I understand.
I will contact yuten-san directly about this issue.

Yes, we should upgrade.
We will address that in a separate ticket!
Thanks 👍